### PR TITLE
# Categories Crud Operation Done

### PR DESCRIPTION
**Pull request by Umadevi04**

### CODE REVIEW

File: app/Http/Controllers/CategoryController.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  \App\Models\Category  $category
+     * @return \Illuminate\Http\Response
+     */     
+
+    public function edit(Category $category)
+    {        
+        $categories=  Category::pluck('name', 'id');
```
Comment (Tamiltuckker) [inline app/Http/Controllers/CategoryController.php]: Don't pluck all categories.Kindly change this line.
get only corresponding category.

File: app/Http/Controllers/CategoryController.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  
+    public function store(Request $request)
+     {       
+        $request->validate([
+            'name' => 'required',                    
+        ]);  
+    
+        $category = new Category();
+        $category->name = $request->name;
+        $category->slug = Str::slug($request->name);
```
Comment (Tamiltuckker) [inline app/Http/Controllers/CategoryController.php]: Use laravel accessors And Mutators concept in model.

File: app/Http/Controllers/CategoryController.php [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+     {            
+        $request->validate([
+            'name' => 'required',          
+        ]); 
+
+        $category->name = $request->name;
+        $category->slug = Str::slug($request->name);   
+
+        if ($request->active == 'on') {
+            $category->active = 1;
```
Comment (Tamiltuckker) [inline app/Http/Controllers/CategoryController.php]: Use model constant.

File: app/Http/Controllers/CategoryController.php [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        } else {
+            $category->active = 0;
```
Comment (Tamiltuckker) [inline app/Http/Controllers/CategoryController.php]: Use model constant.

File: app/Http/Controllers/CategoryController.php [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request,Category $category)
+     {            
+        $request->validate([
+            'name' => 'required',          
+        ]); 
+
+        $category->name = $request->name;
+        $category->slug = Str::slug($request->name);
```
Comment (Tamiltuckker) [inline app/Http/Controllers/CategoryController.php]: @Umadevi04 Use laravel accessors And Mutators concept in model.

File: resources/views/admin/categories/edit.blade.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                                @csrf
+                                @method('PUT')
+                                <div class="col-md-6">
+                                    <div class="form-group">
+                                        <label for="example-text-input" class="form-control-label">Category
+                                            name</label>
+                                        <input class="form-control" name="name" type="text"
+                                            value="{{ old('name', $category->name) }}">
+                                    </div>
+                                </div><br>
```
Comment (Tamiltuckker) [inline resources/views/admin/categories/edit.blade.php]: Remove br tag.

File: resources/views/admin/categories/edit.blade.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                                <div class="col-md-6">
+                                    <div class="form-group">
+                                        <label class="form-check-label" for="flexSwitchCheckDefault">Status</label>
+                                        <div class="form-check form-switch ">
+                                            <input type="hidden" value="{{ $category->id }}" name="categoryid">
+                                            @if ($category->active == 1)
```
Comment (Tamiltuckker) [inline resources/views/admin/categories/edit.blade.php]: Use model constant.

File: resources/views/admin/categories/index.blade.php [hunk 8]
```diff
@@ -0,0 +1,101 @@
+@extends('layouts.app')
+
+@section('content')l
```
Comment (Tamiltuckker) [inline resources/views/admin/categories/index.blade.php:3]: Why you add like @section('content') after -  l - kindly remove it.

File: resources/views/admin/categories/index.blade.php [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        <div class="row">
+            <div class="col-12">
+                <div class="card mb-4">
+                    <div class="card-header pb-0">
+                        @if ($message = Session::get('success'))
+                          @if($message=="Category deleted successfully")                          
+                            <p class="alert alert-danger text-white" role="alert">{{ $message }}</p>     
+                          @else
+                            <p class="alert alert-success text-white" role="alert">{{ $message }}</p>                    
+                          @endif
```
Comment (Tamiltuckker) [inline resources/views/admin/categories/index.blade.php:15]: Change these lines. kindly use it laravel session - >success message after redirection.

File: resources/views/admin/categories/index.blade.php [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+                                        <tr>
+                                            <td>
+                                                <div class="d-flex px-2 py-1">
+                                                    <div class="d-flex flex-column justify-content-center">
+                                                        <h6 class="mb-0 text-sm">{{ $category->name }}</h6>
+                                                    </div>
+                                                </div>
+                                            </td>
+                                            <td class="align-middle text-center text-sm">
+                                                @if ($category->active == 1)
```
Comment (Tamiltuckker) [inline resources/views/admin/categories/index.blade.php]: use Model constant

**Comment by Umadevi04** [comment]
All the corrections are made as you told.

**Comment by Tamiltuckker** [comment]
Great !

**Pull request closed without merging**